modules/metrics-exporter/src/main/java/org/opencastproject/metrics/impl/MetricsExporter.java

```diff
@@ -226,8 +232,17 @@ public void setOrganizationDirectoryService(OrganizationDirectoryService organiz
     this.organizationDirectoryService = organizationDirectoryService;
   }
 
-  @Reference
+  @Reference(
+      policy = ReferencePolicy.DYNAMIC,
+      cardinality = ReferenceCardinality.OPTIONAL,
+      unbind = "unsetAssetManager"
+  )
   public void setAssetManager(AssetManager assetManager) {
     this.assetManager = assetManager;
```

Comment:
On a worker, this now shows:
```
...
jvm_memory_pool_collection_init_bytes{pool="G1 Survivor Space",} 0.0
# HELP opencast_asset_manager_events Events in Asset Manager
# TYPE opencast_asset_manager_events gauge
# HELP process_cpu_seconds_total Total user and system CPU time spent in seconds.
...
```

I'm wondering if it would work to initialize the asset manager metric in here as well:
```suggestion
    this.assetManager = assetManager;
    eventsInAssetManager = Gauge.build()
            .name("opencast_asset_manager_events")
            .help("Events in Asset Manager")
            .labelNames("organization")
            .register();
```

plus unregistering it in `unsetAssetManager()` and not making the class var final.

Reply:
I changed the code as suggested. Now the asset manager metrics are no longer registered unnecessarily on non-admin nodes (tested with worker).